astropy/coordinates/builtin_frames/icrs_observed_transforms.py

```diff
@@ -97,3 +97,14 @@ def observed_to_icrs(observed_coo, icrs_frame):
         icrs_srepr = newrepr.represent_as(SphericalRepresentation)
 
     return icrs_frame.realize_frame(icrs_srepr)
+
+
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, AltAz, AltAz)
+@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, HADec, HADec)
+def observed_to_observed(from_coo, to_frame):
+    if from_coo.is_equivalent_frame(to_frame):  # loopback to the same frame
+        return to_frame.realize_frame(from_coo.data)
+
+    # for now we just implement this through CIRS to make sure we get everything
```

Comment:
minor tweak - this should be ICRS. Could you also add a note explaining why we don't use CIRS, e.g link to this comment explaining why the AA-CIRS-AA route can be slow (https://github.com/astropy/astropy/pull/10994#issuecomment-722617041)

Reply:
done